docs/concept/property-repair.md

```diff
@@ -0,0 +1,211 @@
+
+## Property Repair
+
+Based on the Merkel tree and Gossip concept, the system can proceed with the Property Repair process.
+This process is scheduled to run on each data node daily at 1:00 AM(it's configurable as `property-background-repair-cron` flag), and follows these steps:
+1. **Select a Group**: The node retrieves a list of Property groups where the number of **replicas is greater than or equal to 2**, and randomly selects one group for repair.
+2. **Query Node List**: Then determines the list of nodes that hold replicas for the selected group and sends the gossip propagation message to those nodes to synchronize the Property data for that group.
+3. **Wait for the Result**: The initiating node waits for the final result of the synchronization process before proceeding.
+
+### Property Synchronize between Two Nodes
+
+When two nodes engage in Property data synchronization, they follow a specific protocol to ensure data consistency.
+Let’s refer to the current node as A and the target node as B. The process is as follows:
+
+1. **(A)Establish Connection**: Node A initiates a **bidirectional streaming connection** with node B to enable real-time, two-way data transfer.
+2. **(A)Iterate Over Shards**: Node A retrieves the list of all Property-related shards for the selected group and processes them one by one.
+3. **(A)Send Merkle Tree Summary**: For each shard, node A reads its Merkle Tree and sends a summary (including root SHA and slots SHA) to node B. 
+This allows B to quickly identify which slots may contain differences.
+4. **(B)Verify Merkle Tree Summary and Respond**: Node B compares the received summary against its own Merkle Tree for the same shard and group:
+   * If the root SHA matches, node B returns an empty slot list, indicating no differences. 
+   * If the root SHA differ, node B checks the slot SHA, identifies mismatched slots, and sends back all relevant leaf node details, including the **slot index**, **entity**, and **SHA value**.
+5. **(A)Compare Leaf Data**: Node A processes the received leaf data and takes the following actions: 
+   * For missing entities (present on B but not on A), A requests the full Property data from B.
+   * For entities present on A, but not on B, A sends the full Property data to B.
+   * For SHA mismatches, A sends its full Property data to B for validation.
+6. **(B)Validate Actual Data**: Node B handles the data as follows: 
+   * For missing entities, B returns the latest version of the data.
+   * For entities present on A, but not on B, B updates its local copy with the data from A.
+   * For SHA mismatches, B uses "last-write-win" strategy. It compares the version numbers. If B’s version is newer, it returns the Property data to A. If A’s version is newer, B updates its local copy and does not return any data. If the versions are the same, it selects the data from the smaller index of the node list; in this case, it would be from A.****
```

Comment:
```suggestion
   * For SHA mismatches, B uses the "last-write-win" strategy. It compares the version numbers. If B’s version is newer, it returns the Property data to A. If A’s version is newer, B updates its local copy and does not return any data. If the versions are the same, it selects the data from the smaller index of the node list; in this case, it would be from A.
```